src/anndata/_io/specs/methods.py

```diff
@@ -391,7 +390,50 @@ def write_basic(
     dataset_kwargs: Mapping[str, Any] = MappingProxyType({}),
 ):
     """Write methods which underlying library handles natively."""
-    f.create_dataset(k, data=elem, **dataset_kwargs)
+    dtype = dataset_kwargs.get("dtype", elem.dtype)
+    f.create_dataset(k, data=elem, **dataset_kwargs, dtype=dtype)
+
+
+def _iter_chunks_for_copy(elem: ArrayStorageType, dest: ArrayStorageType):
+    """
+    Returns an iterator of tuples of slices for copying chunks from `elem` to `dest`.
+
+    * If `dest` has chunks, it will return the chunks of `dest`.
+    * If `dest` is not chunked, we write it in ~100MB chunks or 1000 rows, whichever is larger.
+    """
+    if dest.chunks and hasattr(dest, "iter_chunks"):
```

Comment:
Why the `hasattr` here?

`ArrayStorageType` is just `zarr.Array | h5py.Array`: are there subtypes or a minimum version, or does only one of these types have the method?

Reply:
Only one has the method: https://docs.h5py.org/en/stable/high/dataset.html#h5py.Dataset.iter_chunks and https://zarr.readthedocs.io/en/stable/search.html?q=iter_chunks